src/python.cpp

```diff
@@ -1126,19 +1148,38 @@ SEXP py_to_r(PyObject* x, bool convert) {
 
       case NPY_STRING:
       case NPY_UNICODE: {
-        PyObjectPtr itemFunc(PyObject_GetAttrString(ptrArray, "item"));
-        if (itemFunc.is_null())
+
+        PyObjectPtr nditerArgs(PyTuple_New(1));
+        // PyTuple_SetItem steals reference the array, but it's already wraped
+        // into PyObjectPtr earlier (so it gets deleted after the scope of this function)
+        // To avoid trying to delete it twice, we need to increase its ref count here.
+        PyTuple_SetItem(nditerArgs, 0, (PyObject*)array);
+        Py_IncRef((PyObject*)array);
+
+        PyObjectPtr iter(PyObject_Call(get_np_nditer(), nditerArgs, NULL));
```

Comment:
Does it make sense to pass any flags to nditer, like `"readonly"`.
Do we need to pass something like `order = 'F'`, to ensure we get the elements in the correct order, regardless of how the underlying array buffer is arranged?

Reply:
`'readonly'` is the default for nditer. We could force an ordering, but it would be redundant as we already create a Fortran ordered array earlier.

https://github.com/rstudio/reticulate/blob/70a23d693736e7f82b1bbb484579f2849ce41ddd/src/python.cpp#L1077-L1082